### PR TITLE
feat(mutation placing): Add support for mutating static expression bodied constructors

### LINKED ISSUE #959: Static constructor with expression body syntax throws error
**Describe the bug**
When a class has a static constructor using expression body syntax, the following exception is thrown when running the `dotnet stryker` command:

```
System.ArgumentException: Node to track is not a descendant of the root.
   at Microsoft.CodeAnalysis.SyntaxNodeExtensions.TrackNodes[TRoot](TRoot root, IEnumerable`1 nodes)
   at Stryker.Core.Mutants.MutantOrchestrator.MutateStaticConstructor(ConstructorDeclarationSyntax constructorDeclaration, MutationContext context) in d:\a\1\s\src\Stryker.Core\Stryker.Core\Mutants\MutantOrchestrator.cs:line 137
   at Stryker.Core.Mutants.MutantOrchestrator.Mutate(SyntaxNode currentNode, MutationContext context) in d:\a\1\s\src\Stryker.Core\Stryker.Core\Mutants\MutantOrchestrator.cs:line 104
   at Stryker.Core.Mutants.MutantOrchestrator.MutateExpression(SyntaxNode currentNode, MutationContext context) in d:\a\1\s\src\Stryker.Core\Stryker.Core\Mutants\MutantOrchestrator.cs:line 260
   at Stryker.Core.Mutants.MutantOrchestrator.Mutate(SyntaxNode currentNode, MutationContext context) in d:\a\1\s\src\Stryker.Core\Stryker.Core\Mutants\MutantOrchestrator.cs:line 132
   at Stryker.Core.Mutants.MutantOrchestrator.MutateExpression(SyntaxNode currentNode, MutationContext context) in d:\a\1\s\src\Stryker.Core\Stryker.Core\Mutants\MutantOrchestrator.cs:line 260
   at Stryker.Core.Mutants.MutantOrchestrator.Mutate(SyntaxNode currentNode, MutationContext context) in d:\a\1\s\src\Stryker.Core\Stryker.Core\Mutants\MutantOrchestrator.cs:line 132
   at Stryker.Core.Mutants.MutantOrchestrator.MutateExpression(SyntaxNode currentNode, MutationContext context) in d:\a\1\s\src\Stryker.Core\Stryker.Core\Mutants\MutantOrchestrator.cs:line 260
   at Stryker.Core.Mutants.MutantOrchestrator.Mutate(SyntaxNode currentNode, MutationContext context) in d:\a\1\s\src\Stryker.Core\Stryker.Core\Mutants\MutantOrchestrator.cs:line 132
   at Stryker.Core.Mutants.MutantOrchestrator.Mutate(SyntaxNode currentNode) in d:\a\1\s\src\Stryker.Core\Stryker.Core\Mutants\MutantOrchestrator.cs:line 81
   at Stryker.Core.MutationTest.MutationTestProcess.Mutate() in d:\a\1\s\src\Stryker.Core\Stryker.Core\MutationTest\MutationTestProcess.cs:line 90
   at Stryker.Core.StrykerRunner.RunMutationTest(StrykerOptions options, IEnumerable`1 initialLogMessages) in d:\a\1\s\src\Stryker.Core\Stryker.Core\StrykerRunner.cs:line 85
```

I've added a minimal [repro](https://github.com/jasonwoods-7/StaticConstructor) showing the issue.

**Logs**
Log file attached.
[log-20200225.txt](https://github.com/stryker-mutator/stryker-net/files/4251622/log-20200225.txt)

**Expected behavior**
No exception is thrown and the static constructor is correctly mutated.

**Desktop (please complete the following information):**
 - OS: Windows Server 2012 R2
 - Type of project: .Net Core
 - Framework Version: 3.1
 - Stryker Version: 0.16.0 (beta)

**Additional context**
I can work around the issue by changing the static constructor to use statement body syntax instead of expression body syntax.

### PR DESCRIPTION
**Pull request by jasonwoods-7**
I went ahead and made an attempt at fixing the issue I reported in #959.

This is my first time contributing to this project, so let me know if I need to add anything else.

Fixes #959

### CODE REVIEW

**Comment by Mobrockers** [comment]
Thank you for your contribution! We'll review your fix and get back to you if we have any concerns :)

File: src/Stryker.Core/Stryker.Core/Mutants/MutantOrchestrator.cs [hunk 1]
```diff
@@ -134,11 +134,19 @@ private SyntaxNode Mutate(SyntaxNode currentNode, MutationContext context)
 
         private SyntaxNode MutateStaticConstructor(ConstructorDeclarationSyntax constructorDeclaration, MutationContext context)
         {
-            var trackedConstructor = constructorDeclaration.TrackNodes(constructorDeclaration.Body);
-            var mutatedBlock = (BlockSyntax) Mutate(constructorDeclaration.Body, context);
-            var markedBlock = MutantPlacer.PlaceStaticContextMarker(mutatedBlock);
+            var trackedConstructor = constructorDeclaration.TrackNodes((SyntaxNode) constructorDeclaration.Body ?? constructorDeclaration.ExpressionBody);
+            if (constructorDeclaration.ExpressionBody != null)
+            {
+                var markedBlock = Mutate(constructorDeclaration.ExpressionBody, context);
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core/Mutants/MutantOrchestrator.cs]: The method `MutantPlacer.PlaceStaticContextMarker` should be called here. We mark static code so the coverage analyse can handle the static method.

File: src/Stryker.Core/Stryker.Core.UnitTest/Mutants/MutantOrchestratorTests.cs [hunk 2]
```diff
@@ -500,6 +501,29 @@ public void MutationsShouldHaveLinespan()
             mutants.ShouldHaveSingleItem().Mutation.OriginalNode.GetLocation().GetLineSpan().StartLinePosition.Line.ShouldBe(2);
         }
 
+        [Theory]
+        [InlineData(@"static string Value { get; }
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core.UnitTest/Mutants/MutantOrchestratorTests.cs]: This is a lot of lines for an argument. As the class is the same for both arguments you could move that to the method and keep the difference in the arguments.

File: src/Stryker.Core/Stryker.Core/Mutants/MutantOrchestrator.cs (hunk 1 above)
Comment (dupdob) [reply src/Stryker.Core/Stryker.Core/Mutants/MutantOrchestrator.cs]: thanks @richardwerkman, I was going to raise this concern.
I have been thinking about this issue, and the only I could think of was to convert the _ExpressionBody_ to a (_BlockSyntax_) _Body_ so one can add the static marker.

Comment (jasonwoods-7) [reply src/Stryker.Core/Stryker.Core/Mutants/MutantOrchestrator.cs]: I was wondering about that.  Updated.

File: src/Stryker.Core/Stryker.Core.UnitTest/Mutants/MutantOrchestratorTests.cs (hunk 2 above)
Comment (jasonwoods-7) [reply src/Stryker.Core/Stryker.Core.UnitTest/Mutants/MutantOrchestratorTests.cs]: Good point. Updated.

**Pull request merged**